kotlin-example/src/main/kotlin/org/onflow/examples/kotlin/streaming/streamExecutionData/SubscribeExecutionDataExample.kt

```
@@ -0,0 +1,70 @@
+        dataChannel: ReceiveChannel<FlowBlockExecutionData>,
+        errorChannel: ReceiveChannel<Throwable>,
+        receivedExecutionData: MutableList<FlowBlockExecutionData>
+    ) {
+        val dataJob = scope.launch {
+            try {
+                for (data in dataChannel) {
+                    if (!isActive) break // Respond to cancellation
+                    receivedExecutionData.add(data)
+                    yield() // Cooperative cancellation
+                }
+            } catch (e: CancellationException) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)